docs/Lexicon.rst

```diff
@@ -144,6 +144,14 @@ source code, tests, and commit messages. See also the `LLVM lexicon`_.
     themselves. They can be compared across declarations but cannot be used
     directly from within the context.
 
+  irrefutable pattern
+    A pattern that always matches. These patterns either bind to a variable or
+    perform structural modification, e.x.:
+
+    1. ``case _:``.
+    2. ``case let x:``.
+    3. ``case (_, _):``.
```

Comment:
Currently, the compiler only statically accepts tuple patterns when the pattern type is a tuple. Perhaps we could relax that and implicitly insert an `as (_, _)` check, but they are irrefutable in the current model.